pkg/authentication/tokens.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package authentication
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/oauth2"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+)
+
+type JWKSTokenVerifier struct {
```

Comment:
let's add the monitor here as we might want to add metrics re validation

Reply:
Done in both token verifiers